### ISSUE TITLE
`replace_sampler_ddp` doesn't create a shuffled sampler

### ISSUE THREAD

**Issue by ibeltagy** [PyTorchLightning/pytorch-lightning#2703]
## 🐛 Bug

The `DistributedSampler` created using `replace_sampler_ddp` is not shuffled. Check the `kwargs` [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/data_loading.py#L195)

### Expected behavior

If training dataloader, create a shuffled `DistributedSampler`, else create a non-shuffled sampler. Even though the `train` flag is passed to the function [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/data_loading.py#L146), it is ignored. 

### Environment

pytorch-lightning master
